src/query/plan/operator.cpp

```diff
@@ -5996,24 +6002,47 @@ class PeriodicCommitCursor : public Cursor {
   PeriodicCommitCursor(const PeriodicCommit &self, utils::MemoryResource *mem)
       : self_(self), input_cursor_(self.input_->MakeCursor(mem)) {
     MG_ASSERT(input_cursor_ != nullptr, "PeriodicCommitCursor: Missing input cursor.");
+    MG_ASSERT(self_.commit_frequency_ != nullptr, "Commit frequency should be defined at this point!");
   }
 
   bool Pull(Frame &frame, ExecutionContext &context) override {
+    // NOLINTNEXTLINE(misc-const-correctness)
     OOMExceptionEnabler oom_exception;
     const SCOPED_PROFILE_OP_BY_REF(self_);
 
-    throw utils::NotYetImplemented("periodic commit");
+    if (!commit_frequency_ == -1) [[unlikelly]] {
```

Comment:
You are comparing here optional without value with -1. This is a mistake.

```suggestion
    if (!commit_frequency_.has_value()) [[unlikelly]] {
```